### PR TITLE
Grid Visualizer: Improve observation logic

### PR DESCRIPTION
**Pull request by t-hamano**
Fix the issue found in this comment: https://github.com/WordPress/gutenberg/pull/64425#pullrequestreview-2519538374

## What?

This PR makes the grid visualizer's frames match their actual child elements more accurately.

## Why?

This component is used to draw the frames for the blocks inside it when the block is a grid layout type. This frame is re-rendered when the grid block or its children change size. However, when padding or borders are applied to a grid block, this frame may not match the actual child blocks:

image

## How?

To resolve this mismatch, I made two changes:

- On the grid element itself, add a `MutationObserber` to monitor changes to style attributes, so that the frame is re-rendered when, for example, the vertical padding changes.
- In the `getGridInfo()` function, include the border width for padding calculations.

## Testing Instructions

- Insert a Grid block.
- Change the styles for padding, margin, gap, border, etc.
- The child elements and visualizer positions should match perfectly.
- This logic should also work correctly when the grid block has no "box-sizing: border-box" style applied. Remove [this style](https://github.com/WordPress/gutenberg/blob/212144c8dd868e8f0faab851aadb0aa06a38de3b/packages/block-library/src/group/style.scss#L3) and test again.
- It should work correctly when applied via global styles, not just on the block instance.
  - **Note:** However, only the vertical padding does not sync immediately in the global styles, probably because neither the `ResizeObserver` nor the `MutationObserber` can detect the change. If you have an approach to solve this, please let me know.

## Screenshots or screencast 

https://github.com/user-attachments/assets/4b6cfb5e-7228-40f2-9aaf-f3d677885d73

### CODE REVIEW

File: packages/block-editor/src/components/grid/grid-visualizer.js [hunk 1]
```diff
@@ -62,6 +62,17 @@ const GridVisualizerGrid = forwardRef(
 				observer.observe( element );
 				observers.push( observer );
 			}
+
+			const mutationObserver = new window.MutationObserver( () => {
+				setGridInfo( getGridInfo( gridElement ) );
+			} );
+			mutationObserver.observe( gridElement, {
+				attributeFilter: [ 'style' ],
```
Comment (talldan) [inline packages/block-editor/src/components/grid/grid-visualizer.js]: I wonder if 'class' should also be added to this list. Change block style variations or adding a custom classname could also result in a change of top padding/border.

**Review by talldan — APPROVED** [review]
Thanks for fixing this.

This seems to be working from well from my testing.

I also notice that it's mostly seems to be top padding and top border that cause the issues, they don't cause a resize so the resize observer isn't being triggered.

**Comment by t-hamano** [comment]
Thanks for the PR!

Sorry, I forgot to attach the video, so I uploaded it.

> I wonder if 'class' should also be added to this list.

I think this is right, so I would like to add it.

**Comment by stokesman** [comment]
> **Note:** However, only the vertical padding does not sync immediately in the global styles, probably because neither the `ResizeObserver` nor the `MutationObserber` can detect the change. If you have an approach to solve this, please let me know.

@t-hamano, just letting you know #68842. I’d appreciate your 👀 on that one 🙇.

**Pull request merged**